Common/Securities/IBuyingPowerModel.cs

```diff
@@ -37,6 +37,27 @@ public interface IBuyingPowerModel
         /// <param name="leverage">The new leverage</param>
         void SetLeverage(Security security, decimal leverage);
 
+        /// <summary>
+        /// Gets the margin currently allocated to the specified holding
+        /// </summary>
+        /// <param name="security">The security to compute maintenance margin for</param>
+        /// <returns>The maintenance margin required for the </returns>
+        decimal GetMaintenanceMargin(Security security);
+
+        /// <summary>
+        /// The margin that must be held in order to increase the position by the provided quantity
+        /// </summary>
+        /// <param name="security">The security to compute initial margin for</param>
+        /// <param name="quantity">The change in the contemplated change in shares</param>
+        decimal GetInitialMarginRequirement(Security security, decimal quantity);
```

Comment:
Not sure if worth it but following the pattern of the methods below and for future proof, wondering if we should make these new methods return a `Result` class type and receive a `Parameters` class type? Maybe the result class isn't as needed but the parameters one might? 🤔

Reply:
I had considered this as well --if we're confident that the number of potential broken users is near-zero then I think this is a good move -- the concern is if users do have to make this change, before it's just `protected` -> `public` -- but if we're changing the signatures/etc then that might be a little more difficult.

Reply:
Could add something to `BuyingPowerModel` that keeps the protected virtual and checks for overrides and provides an `[Obsolete]` error message via compile?

Reply:
> I had considered this as well --if we're confident that the number of potential broken users is near-zero then I think this is a good move 

I think so, because since these methods were protected only usages will be custom buying power models really which already would get broken by the `protected` -> `public` change?